reconcile/utils/oc.py

```diff
@@ -1016,16 +1134,24 @@ def __init__(
             init_api_resources=False,
             local=local,
             insecure_skip_tls_verify=insecure_skip_tls_verify,
+            connection_parameters=connection_parameters,
         )
 
+        if connection_parameters:
+            token = connection_parameters.automation_token
+            if connection_parameters.is_cluster_admin:
+                token = connection_parameters.cluster_admin_automation_token
```

Comment:
This logic is similar to https://github.com/app-sre/qontract-reconcile/pull/3176/files#diff-006493679cf46d27e100feeb8475e6e29987fe85cca4858a1f05608b2d536dc8R382-R387, but has different conditions, maybe extract to a method like `get_token` and reuse it?

Reply:
Yeah the conditions differ - I basically translated the original logic. I'd like to keep the logic as similar as possible, as I'm not 100% sure about all the use-cases or why the condition was implemented that way originally. Now the question is: which token you want to extract with `get_token()`? The admin token or the non-admin token?